addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3449,36 +3566,17 @@ class SnippetsMenu extends Component {
                 this.options.wysiwyg.odooEditor.observerUnactive('dragAndDropCreateSnippet');
 
                 dropped = false;
-                const snippetId = element.closest('.oe_snippet').dataset.oeSnippetId;
-                const snippet = this.snippets.get(parseInt(snippetId));
+                const snippetKey = element.closest('.oe_snippet').dataset.snippetKey;
+                const snippet = this.snippets.get(parseInt(snippetKey));
                 $snippet = $(element).closest('.oe_snippet');
-                const $baseBody = $(snippet.baseBody);
-                var $selectorSiblings = $();
-                var $selectorChildren = $();
-                const selectorExcludeAncestor = [];
-                var temp = this.templateOptions;
-                for (var k in temp) {
-                    if ($baseBody.is(temp[k].base_selector) && !$baseBody.is(temp[k].base_exclude)) {
-                        if (temp[k]['drop-near']) {
-                            $selectorSiblings = $selectorSiblings.add(temp[k]['drop-near'].all());
-                        }
-                        if (temp[k]['drop-in']) {
-                            $selectorChildren = $selectorChildren.add(temp[k]['drop-in'].all());
-                        }
-                        if (temp[k]['drop-exclude-ancestor']) {
-                            selectorExcludeAncestor.push(temp[k]['drop-exclude-ancestor']);
-                        }
-                    }
-                }
 
-                // Prevent dropping an element into another one.
-                // (E.g. ToC inside another ToC)
-                for (const excludedAncestorSelector of selectorExcludeAncestor) {
-                    $selectorSiblings = $selectorSiblings.filter((i, el) => !el.closest(excludedAncestorSelector));
-                    $selectorChildren = $selectorChildren.filter((i, el) => !el.closest(excludedAncestorSelector));
-                }
+                const $baseBody = $(snippet.baseBody);
+                const selectors = this._getSelectors($baseBody);
+                const $selectorSiblings = selectors.$selectorSiblings;
+                const $selectorChildren = selectors.$selectorChildren;
```

Comment:
Nitpick, but you could directly extract `const { $selectorSiblings, $selectorChildren }`